dbt/include/databricks/macros/adapters.sql

```diff
@@ -111,6 +111,37 @@
   {% endif %}
 {% endmacro %}
 
+{% macro optimize(relation) %}
+  {{ return(adapter.dispatch('optimize', 'dbt')(relation)) }}
+{% endmacro %}
```

Comment:
At first I wondered why these lines are here. It seems to bring the `dbt.optimize` macro into the current namespace so you can reference it on line 131?

For my own understanding: Jinja supports importing macros similar to Python (`{% import "../path/to/file" as <importable-name> %}`), but it looks like dbt doesn't use that syntax. Instead we "import" functions by making local macros that call `adapter.dispatch`. Does that sound correct?

Reply:
this is some dbt magic to allow for inheritance and user overwrites.